erpnext/crm/doctype/opportunity/opportunity.js

```diff
@@ -116,6 +123,19 @@ frappe.ui.form.on("Opportunity", {
 		}
 	},
 
+	service_fee: function (frm) {
+		frm.trigger("calculate_opportunity_cost");
+	},
+
+	software_fee: function (frm) {
+		frm.trigger("calculate_opportunity_cost");
+	},
+
+	calculate_opportunity_cost: function (frm) {
+		let amount = frm.doc.service_fee + frm.doc.software_fee;
```

Comment:
Use `flt` for float conversions here, just in case it's not the type we expect. If it's a non-float type, Frappe will set it to 0.

```suggestion
		let amount = flt(frm.doc.service_fee) + flt(frm.doc.software_fee);
```